### ISSUE TITLE
Prevent dropdown items filter

### ISSUE THREAD

**Issue by Everduin94** [yairEO/tagify#1098]
## Prerequisites

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Goal

To clarify, the example below does not use tagify, I'm trying to replicate this behavior using tagify.

The initial dropdown list shows all possible values and does not filter based on what is already selected. 

In the example below, I've selected all 3 tags previously. When I click the component, it shows all 3 tags, even though they're already selected.

<img width="674" alt="image" src="https://user-images.githubusercontent.com/14320878/184550310-883f2907-d557-4f47-9551-98f5bb7c25a2.png">

Given input, then filter the dropdown.

<img width="685" alt="image" src="https://user-images.githubusercontent.com/14320878/184550586-e51379c8-e123-4fed-b0c0-a0c5f84aab55.png">

### What I Tried so Far

Looking through the documentation and previous issues, I didn't see any examples or anyone requesting something similar.

Using the `duplicates: true` setting seems to get me close (the list then has all possible options), but I don't actually want to allow duplicates. 

I've tried manually setting the suggestions and suggestListItems on open. But this doesn't seem to actually change what's in the list. Maybe the dropdown is already rendered at this point and changing it on show doesn't effect its behavior?

```javascript
this.tagify.on('dropdown:show', (e) => {
    this.tagify.state.dropdown.suggestions = this.tagify.whitelist;
    this.tagify.suggestedListItems = this.tagify.whitelist;
});
```

---

Also just a side note. This is a great project and I really enjoying using it.

**Comment by yairEO** [comment]
Thanks for the nice side note :) I have been working on Tagify constantly for years for this reason.

1. Why would you want to show already-selected items in the suggestions dropdown if you don't allow them to be added?
2. Do you ever want to filter the suggestions-dropdown list, at all?

The `dropdown:show` event is indeed called *after* the dropdown has been rendered.

**Comment by Everduin94** [comment]
1. I plan to render an inline button in the dropdown item (presumably through the `templates` prop). So referring to the original picture, you can see the "..." on the right side of the item. My app does the same, again this isn't tagify, this is what I want to replace with tagify.

<img width="798" alt="image" src="https://user-images.githubusercontent.com/14320878/184556207-5ce2c95b-75ea-421a-a37d-d35345613e82.png">

Notice how "Web Dev" is selected, but also available in the drop down, giving me the ability to click its corresponding edit button "...".

2. Yes, I want to see all the tags initially (so to point 1, you could edit any tag regardless if it's selected or not). But if you type, it filters based on that input. 
- in other words, **only filter based on input** and **not selected tags + input**.

<img width="629" alt="image" src="https://user-images.githubusercontent.com/14320878/184556402-9fddec07-b2e4-4a95-9a26-7cfff305bcfb.png">

Also, just another positive note lol. The existing tag component I use is the Material chip/autocomplete component and tagify is significantly better imo.

**Comment by yairEO** [comment]
Tagify is at a whole other level than any other component in terms of built-in support for tons of scenarios, and highly customizable for unique cases, such as this.

I will make a new setting for such scenario, called `dropdown.includeSelectedTags` (which will be `false` by default)
